packages/list_bloc/lib/src/cubits/list_cubit.dart

```diff
@@ -20,5 +19,58 @@ class ListCubit<T, F> extends DataCubit<List<T>, F> {
       rethrow;
     }
   }
-// TODO(alexis): implement remove, add, replace etc
+
+  void add(T item) {
+    try {
+      final newData = <T>[...state.data ?? [], item];
+      emit(Data(data: newData, filter: state.filter));
+    } catch (e) {
+      emit(Data.error(data: state.data, filter: state.filter, error: e));
+      rethrow;
+    }
+  }
+
+  void removeAt(int index) {
+    try {
+      final newData = List<T>.from(state.data ?? [])..removeAt(index);
+      emit(newData.isEmpty ? Data.empty(filter: state.filter) : Data(data: newData, filter: state.filter));
+    } catch (e) {
+      emit(Data.error(data: state.data, filter: state.filter, error: e));
+      rethrow;
+    }
+  }
+
+  void remove(T item) {
+    try {
+      final newData = List<T>.from(state.data ?? [])..remove(item);
+      emit(newData.isEmpty ? Data.empty(filter: state.filter) : Data(data: newData, filter: state.filter));
```

Comment:
@Groseuros Can we make this `isEmpty` check DRY? It will be better to do so as we can avoid repeating same code.

Reply:
@adar2378 You are right! I updated the PR and did the same with Data.error, following the same logic